cdxev/amend/operations.py

```diff
@@ -132,11 +132,13 @@ class InferSupplier(Operation):
     def handle_component(
         self, component: dict, path_to_license_folder: str = ""
     ) -> None:
-        if (
-            ("supplier" in component)
-            or ("publisher" in component)
-            or ("author" in component)
-        ):
+        if "supplier" in component:
+            return
+        if "publisher" in component:
+            component["supplier"] = component["publisher"]
```

Comment:
This assignment should only happen if supplier empty, we should not delete/change information (this would be also according to your documentation, where you state that "if a component does *not* have a supplier".)